### PR TITLE
In resume instruction, update StackLimits from generated code

### PR DESCRIPTION
**Pull request by frank-emrich**
This PR is part of a series that implements native stack switching.

This particular PR moves one particular aspect out of the `tc_resume` libcall: Updating the `StackLimits` object of the parent of the continuation being resumed.

Doing this *inside* the libcall had practical reasons: We needed access to the updated `last_wasm_exit_pc`/`last_wasm_exit_fp`   value in the `VMRuntimeLimits` in order to copy them into the `StackLimits`. However, these values are only updated by the libcall mechanism itself, so they are only available inside the libcall implementation itself.

This PR obtains the corresponding values using code generated for `resume`, before the actual `tc_resume` libcall happening, and writes them into the `StackLimits`.

1. We use the `get_frame_pointer` instruction to obtain a value for `last_wasm_exit_fp`.
2. The value for `last_wasm_exit_pc` is obtained using a new CLIF instruction, `get_instruction_pointer`. All this does is giving us some instruction pointer that is guaranteed to be associated with the current Wasm instruction being translated (i.e., `resume` in our case). While this means that we will write slightly different values for `last_wasm_exit_pc` into the `StackLimits` than before, this difference does not matter at all for backtrace creation. `last_wasm_exit_pc` is never used for control flow (i.e., it is never branched to), all that matters is what Wasm instruction it is associated with.

I consider this to be a workaround, once native stack switching is fully rolled out it would be nice to overhaul the whole backtrace generation mechanism in the longer term. But this PR's goal is to make it possible to move to native stack switching with as little changes to backtrace creation as possible.

### CODE REVIEW

**Review by dhil — COMMENTED** [review]
LGTM, but I have a question regarding `GetRip`.

File: cranelift/codegen/src/isa/x64/inst/emit.rs [hunk 1]
```diff
@@ -4348,6 +4348,15 @@ pub(crate) fn emit(
         Inst::DummyUse { .. } => {
             // Nothing.
         }
+
+        Inst::GetRip { dst } => {
```
[the remark this answers is not in the document]
Comment (frank-emrich) [reply cranelift/codegen/src/isa/x64/inst/emit.rs:4352]: The name of the CLIF instruction is `get_instruction_pointer`. What you are looking at here is a value of type `MInst` within the `x64` backend, which `get_instruction_pointer` gets lowered to (on `x64`). I haven't implemented the instruction on any other platform.

[the remark this answers is not in the document]
Comment (dhil) [reply cranelift/codegen/src/isa/x64/inst/emit.rs:4352]: OK, then the name `GetRip` makes sense, but what about `get_instruction_pointer` then? Do we ever need to do anything else than reading it? Maybe `read_program_counter` is a better name then?

[the remark this answers is not in the document]
Comment (frank-emrich) [reply cranelift/codegen/src/isa/x64/inst/emit.rs:4352]: I don't have strong feelings about renaming it to `get_program_counter`, though.

[the remark this answers is not in the document]
Comment (frank-emrich) [reply cranelift/codegen/src/isa/x64/inst/emit.rs:4352]: Are you fine with the current state then? I would like to land this

[the remark this answers is not in the document]
Comment (dhil) [reply cranelift/codegen/src/isa/x64/inst/emit.rs:4352]: Yes, I am fine.

**Pull request merged**